### DISCUSSION TITLE
Locked out password

### DISCUSSION THREAD

**Discussion by MikeyBeez** [open-webui/open-webui#1027] [General]
# Bug Report

## Description
When I try to log in it shows my user name but asks for my password.  Then it says it wants my email.  Then it says the password is invalid.  Can I delete my user or email address and start over?  I deleted the container, but that didn't help.  

**Bug Summary:**
[Provide a brief but clear summary of the bug]

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

**Expected Behavior:**
[Describe what you expected to happen.]

**Actual Behavior:**
[Describe what actually happened.]

## Environment

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I have reviewed the troubleshooting.md document.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by justinh-rahb** [comment]
> [!CAUTION]
> This will delete your accounts and chat history database, uploaded documents, etc.

Delete the data volume to start over:
```
docker volume rm open-webui
```

Be sure this is your only option... but maybe it's not, how did you end up in this situation? I assume it _**was**_ working at some point? What changed between then and now. I'd suggest reading the docs also: https://docs.openwebui.com

**Reply by tbendien** [reply]
A suggestion is that the admin can set up a second admin user ID as a backup with a different password.

Im also wondering how you can reset your admin password if you get locked out/forget it?

**Reply by justinh-rahb** [reply]
Without another admin account, you can't.

**Reply by duhow** [reply]
Isn't it possible to modify the SQLite file and set a known hash password (like `admin`)?

**Reply by justinh-rahb** [reply]
> Isn't it possible to modify the SQLite file and set a known hash password (like `admin`)?

Of course, up to your own risk tolerance if you want to attempt DB surgery though. There's no instructions in the docs for doing so, nor any other destructive commands that could result in data loss, therefore nobody can blame the project for losing their data.

**Reply by apeckham** [reply]
Notes on attempting db surgery to reset a password:

- `docker exec -ti open-webui bash`
- `apt-get update && apt-get install sqlite3`
- `sqlite3 data/ollama.db`
- `update auth set password='HASH' where email='x@y.com'`

Where HASH is the output of `htpasswd -bnBC 10 "" your-new-password | tr -d ':\n'` (see [utils.py](https://github.com/open-webui/open-webui/blob/main/backend/utils/utils.py))

Thank you for open-webui 🙏

**Reply by jfromaniello** [reply]
thanks @apeckham this worked for me. 

In my case the db is not `data/ollama.db` but `data/webui.db` though

**Reply by justinh-rahb** [reply]
> thanks @apeckham this worked for me.
> 
> In my case the db is not `data/ollama.db` but `data/webui.db` though

Old advice from before our project rename. The database was renamed as well.

**Reply by zacnickens** [reply]
i have removed all volumes in docker and still cant get a new instantiation of openweb-ui. so im stuck with this issue and the removing volumes and the db surgery option havent worked. Is there anything else i should try?

**Reply by wolfverinehim** [reply]
In my case
docker exec -ti open-webui bash
apt-get update && apt-get install sqlite3
sqlite3 data/webui.db
update auth set password='HASH' where email='x@y.com'

htpasswd -bnBC 10 "" your-new-password | tr -d ':\n'
apt update && apt install apache2-utils -y

**Reply by shamong** [reply]
please same issue, can't able to solve, I am running in windows.

**Reply by Aplas99** [reply]
My Solution:
docker exec -ti your-openwebui-instance bash
apt-get update && apt-get install sqlite3
sqlite3 data/webui.db
update auth set password='HASH' where email='your-admin@email.com'
CRTL+D/exit sqlite terminal

docker run -it --rm -v open-webui:/your-backend alpine
apk add apache2-utils sqlite
htpasswd -bnBC 10 "" your-new-password | tr -d ':\n'
copy the hash
sqlite3 /your-backend/webui.db
UPDATE auth SET password='$your-hash' WHERE email='your-admin@email.com';
Done!

**Comment by MikeyBeez** [comment]
Thank you so much!  I love your software.

**Comment by justinh-rahb** [comment]
### Troubleshooting "Wrong Password" Errors in Docker Installations

**Overview:**

"Wrong password" errors typically fall into two categories. Here's how to identify and resolve them:

**1. Migration Issue from Ollama WebUI to Open WebUI:**

- **Problem**: Initially installed as **Ollama WebUI** and later instructed to install **Open WebUI** without seeing the migration guidance. This leads to two docker installations: `ollama-webui` and `open-webui`, each with their own persistent volumes sharing names with their containers. The new volume creation for `open-webui` results in a "wrong password" error upon first login attempt after upgrade, missing previous accounts and chat histories.
- **Solution**: Your original data volume remains intact unless deleted (which is not advised in our guides). To resolve:
  - Follow the migration guide at [Open WebUI Migration](https://docs.openwebui.com/migration).
  - Alternatively, rerun your docker run command with the volume mount `-v ollama-webui:/app/backend/data` instead of `-v open-webui:/app/backend/data` to restore your old data.

**2. Persistent Volume or Installation Method Issues:**

- **Problem**: Launched the docker container without a persistent volume for `/app/backend/data`, or alternated between Docker and manual installations, leading to data being saved in a new location and starting anew.
- **Solution**: Ensure a persistent volume is used for `/app/backend/data` during Docker container setup to maintain data continuity across installations, use `docker cp` command to copy data to/from Docker containers/volumes and your host filesystem.

By addressing these issues accordingly, most "wrong password" errors are resolved. If these solutions don't work, you might be in a different scenario.

For intact data with a truly forgotten password, consider manual SQLite database modification as mentioned above—this requires knowledge or careful research on your own time to avoid potential data loss.

**Comment by lic9425** [comment]
I also forgot my password (more likely I entered it wrong when registering). Why didn't these excellent developers use a second password confirmation on the registration interface? When I tried to re-register, it prompted: The email address has been used. After trying to register and log in with a new email address, it prompted again: You are not an administrator. 😂

**Comment by lmaddox** [comment]
I've had to do this on a near daily basis

**Comment by Wusaru** [comment]
sqlite3 data/webui.db
UPDATE auth SET password='bcrypt hash from cyberchef' WHERE email='x@y.com';

**Comment by ahmed-azraq** [comment]
I am using mac, it gives me this error "Error: unable to open database "data/webui.db": unable to open database file"

**Reply by 3n3v3n** [reply]
same here

**Reply by jloveland** [reply]
you need to be exec into the container.

**Comment by Rjlintkh** [comment]
Also had the issue today after updating open-webui. I am using Docker on Windows, I ran
```
sqlite3 data/webui.db "SELECT * FROM auth;"
```
and found that my username was 'admin@localhost' (which i guessed many many times and found that I wasnt even close)
I followed the guide [https://docs.openwebui.com/troubleshooting/password-reset/#step-2-update-the-password-locally-](url) but I think the escaping on Windows might be a little different, because it still said password incorrect.
So I did the following:
```
htpasswd -bnBC 10 "" alpine | tr -d ':\n' | sqlite3 data/webui.db "UPDATE auth SET password = '$(cat)' WHERE email = 'admin@localhost';"
```
to safely set the password to `alpine`. Hope this helps others too.

**Reply by beriksson3** [reply]
This worked for me with a local installation on Ubuntu 24.04 :)

**Comment by gregruthenbeck** [comment]
I found that my hash contained characters (`$`) that I wasn't escaping properly when using bash commands to try to update the auth record in the DB. So going into the container's terminal, then using an SQLite3 session worked:

1. `docker exec -it open-webui /bin/bash` to open a terminal inside the container
2. `htpasswd -bnBC 10 "" your-password` (without the `| tr -d ':\n') so that you can be sure you're seeing the entire hash.
3. `sqlite3 data/webui.db` to enter the SQLite3 DB session.
4. `SELECT * FROM auth;` to see your registered email.
5. `UPDATE auth SET password='$2...BIG-LONG-HASH...Yu..or something' WHERE email='you@email.com';` to update the password.
6. `SELECT * FROM auth;` to see your updated hash is the same as what you want to set it to.
7. `.exit` to close the DB session.

**Comment by sitorender** [comment]
What worked for me on Windows11, Python installation:

i've downloaded httpd-2.4.66-251206-Win64-VS17.zip and sqlite-tools-win-x64-3510100.zip and extracted them in a folder (for example C:)

In cmd as admin i've run
C:\>cd C:\sqlite-tools-win-x64-3510100
C:\sqlite-tools-win-x64-3510100>sqlite3.exe "C:\pathtoopenwebui\openwebui\open-webui-venv\Lib\site-packages\open_webui\data\webui.db" "SELECT email FROM auth;"

itgivesyourcurrent@email

C:\sqlite-tools-win-x64-3510100>cd C:\httpd-2.4.66-251206-Win64-VS17\Apache24\bin
C:\httpd-2.4.66-251206-Win64-VS17\Apache24\bin>htpasswd -bnBC 10 "" "yournewpassword"

:$2y$10ANDABUNCHOFOTHERLETTERSAND832479823

C:\httpd-2.4.66-251206-Win64-VS17\Apache24\bin>cd C:\sqlite-tools-win-x64-3510100
C:\sqlite-tools-win-x64-3510100>sqlite3.exe "C:\pathtoopenwebui\openwebui\open-webui-venv\Lib\site-packages\open_webui\data\webui.db" "UPDATE auth SET password='$2y$10ANDABUNCHOFOTHERLETTERSAND832479823' WHERE email='itgivesyourcurrent@email';"

(to verify it worked, this should give you |60 or more than 60)
C:\sqlite-tools-win-x64-3510100>sqlite3 "C:\pathtoopenwebui\openwebui\open-webui-venv\Lib\site-packages\open_webui\data\webui.db" "SELECT email, length(password) FROM auth;"
itgivesyourcurrent@email|60
